pkg/execution/runner/runner.go

```diff
@@ -447,9 +447,20 @@ func (s *svc) pauses(ctx context.Context, evt event.Event) error {
 			return err
 		}
 
+		logger.From(ctx).Debug().
+			Str("pause_id", pause.ID.String()).
+			Str("run_id", pause.Identifier.RunID.String()).
+			Msg("consuming pause")
+
+		idx, err := s.state.ConsumePause(ctx, pause.ID, evtMap)
```

Comment:
Moving `consumePause` to above enqueue adds a failure condition which means the function may never be ran:

- If enqueueing fails, the pause is already consumed.
- This means that no new events will trigger the pause, but the step doesn't run.

Can we move this back?  Any particular reason this was moved?

Reply:
If we need the index, shouldn't this be a part of the pause itself vs "whatever the index is now"?  The index is mutated by any ongoing parallel work while this step is paused.

Reply:
Mm was thinking this as we had a similar problem with enqueuing the next step outside of pauses.

The stack only contains resolved work, so the index needs to be the length of the stack as a result of inserting the resolved pause; we can't enqueue the next task before we've mutated the stack. 🤔